django/contrib/auth/views.py

```diff
@@ -442,11 +446,23 @@ class PasswordResetConfirmView(PasswordContextMixin, FormView):
     @method_decorator(never_cache)
     def dispatch(self, *args, **kwargs):
         assert 'uidb64' in kwargs and 'token' in kwargs
+        uidb64 = kwargs['uidb64']
+        token = kwargs['token']
 
         self.validlink = False
-        self.user = self.get_user(kwargs['uidb64'])
+        self.user = self.get_user(uidb64)
 
-        if self.user is not None and self.token_generator.check_token(self.user, kwargs['token']):
+        if token == INTERNAL_RESET_URL_TOKEN:
+            token = self.request.session.get(INTERNAL_RESET_SESSION_TOKEN)
+        else:
+            self.request.session[INTERNAL_RESET_SESSION_TOKEN] = token
+            resolver = self.request.resolver_match
+            resolver_kwargs = resolver.kwargs
+            resolver_kwargs['token'] = INTERNAL_RESET_URL_TOKEN
+            return HttpResponseRedirect(reverse(resolver.url_name,
```

Comment:
I think so, btw please do `resolver.kwargs.copy()` to leave the original kwargs in place on the resolver object.